dbms/src/Storages/Page/V3/Universal/UniversalPageStorage.cpp

```diff
@@ -354,6 +386,8 @@ UniversalPageStorage::dumpIncrementalCheckpoint(const UniversalPageStorage::Dump
         page_directory->copyCheckpointInfoFromEdit(edit_from_mem);
     }
 
+    // FIXME: should push forward the `last_checkpoint_sequence` only when the local files
+    //        are successfully uploaded to remote source
     last_checkpoint_sequence = snap->sequence;
```

Comment:
maybe we should make the upload function a callback in `DumpCheckpointOptions`? @breezewish